lib/install.js

```diff
@@ -382,11 +383,13 @@ Installer.prototype.loadAllDepsIntoIdealTree = function (cb) {
     steps.push([validateArgs, this.idealTree, this.args])
     steps.push([loadRequestedDeps, this.args, this.idealTree, saveDeps, cg.newGroup('loadRequestedDeps')])
   } else {
-    if (this.prod) {
+    if (this.prod && this.dev) {
```

Comment:
Agreed that this is something we want to avoid.

I'll roll back my change, and figure out if there's a better way to fix the bug, without also breaking the `shrinkwrap-dev-depdency.js` test-case.  [Suggestions welcome!]

As far as this functionality goes:

1. Are there cases where we _cannot_ guarantee that transitive devDependencies won't affect the regular/transitive dependencies that get installed?
2. Should this behavior be documented and tested?
3. What about weird stuff like this? 
    
    ```json
    "dependencies": {   
      "lodash": "2.0.0"
    },
    "devDependencies": {
      "lodash": "4.0.0"
    }
    ```
    Currently, we seem to treat this as valid (although many things don't work).  `npm install` against this package resolves `lodash@2.0.0`, and exits cleanly, only indicating that anything is run when `npm ls` is printed to the console.  `npm install --prod` and `npm ls --prod` work as expected.  `npm install --only=dev` doesn't install anything.

That being said, maybe we're starting to stray too far away from the original issue?